### ISSUE TITLE
Local and class variable names resolving to similar package names.

### ISSUE THREAD

**Issue by eliasku** [TiVo/intellij-haxe#95]
when navigating, plugin parses local variables and class variables as package names! If I have `game` package, and later i have class var `game` - plugin resolves var as package. Very uncomfortable. Maybe we have to change the resolving priority or something?

TIR: STB-7315

**Comment by EBatTiVo** [comment]
What are the steps to duplicate this?  Do you have a code example?

If anybody wants to take a look, the resolving code is in ``` HaxeReferenceImpl.resolve() ```. and ```HaxeResolver..resolve() ```.

**Comment by eliasku** [comment]
![image](https://cloud.githubusercontent.com/assets/3038174/6043294/633211de-ac9c-11e4-873b-924746d0d9a3.png)

render/Render.hx
```
package render;

class Render {

	public var a:Int = 0;
	
	public function new() {
	}
}
```

IDETest.hx
```
package ;

import render.Render;

class IDETest {

	public var render:Render;

	public function new() {
		render.a = 0;
	}
}
```

![image](https://cloud.githubusercontent.com/assets/3038174/6043329/99d3365a-ac9c-11e4-92b8-c3e89898836f.png)

**Comment by sganapavarapu1** [comment]
I verified that, this is fixed in TiVo/intellij-haxe repo's ClassHierarchy branch. 
This fix will be available in "0.8.1.1.TiVo.ClassHierarchy.15: (community version, TiVo RC4)" release (coming next week).

**Comment by eliasku** [comment]
Great! Thank you!

**Comment by EBatTiVo** [comment]
Reopening.  Occurring again in Issue #176.

**Comment by giabao** [comment]
Hi, I think there are many issues related to function [HaxeResolver#resolve](https://github.com/TiVo/intellij-haxe/blob/4c8fa986d83f868bf61ee2faecec7ab9ba0fa0cc/src/com/intellij/plugins/haxe/lang/psi/HaxeResolver.java#L50).
Can you ( @fkorotkov  @as3boyan @EBatTiVo @sganapavarapu1 ) explain this function (maybe comment right in the source code). How it relate to [Resolution Order](http://haxe.org/manual/type-system-resolution-order.html)?

**Comment by EBatTiVo** [comment]
Fixed PR #406: https://github.com/TiVo/intellij-haxe/commit/d44634bceb2c799e77714ddbdea44818284b1c1d
